Tools/debug/CollectSystemInfo.ps1

```diff
@@ -88,6 +88,75 @@ function Export-EventLogs {
     }
 }
 
+function Ensure-DiskSpd {
+    $scriptDir = $null
+    try {
+        $scriptDir = Split-Path -Parent $MyInvocation.MyCommand.Definition
+    } catch {}
+    if (-not $scriptDir -or $scriptDir -eq '' -or -not (Test-Path $scriptDir)) {
+        $scriptDir = (Get-Location).Path
+    }
+
+    $diskspdExe = Join-Path $scriptDir 'diskspd.exe'
+    $zipPath = Join-Path $scriptDir 'DiskSpd.ZIP'
+    $extractDir = Join-Path $scriptDir 'DiskSpdExtracted'
```

Comment:
Script is part of ISO, so $scriptDir is a read-only directory and you CAN'T extract files to it!